### PR TITLE
Navigation

### PR DESCRIPTION
**Pull request by Patowhiz**

### CODE REVIEW

**Comment by Patowhiz** [comment]
Fixes #10

File: ClimsoftVer4/ClimsoftVer4/ucrNavigation.vb [hunk 1]
```diff
@@ -1,4 +1,90 @@
 ﻿Public Class ucrNavigation
+    Private bFirstLoad As Boolean = True
+    Public maxRows As Integer
+    Public currRow As Integer
+
+    Public Overrides Sub PopulateControl()
+        MyBase.PopulateControl()
+        currRow = 0
+        maxRows = dtbRecords.Rows.Count
```
Comment (maxwellfundi) [inline ClimsoftVer4/ClimsoftVer4/ucrNavigation.vb:9]: Does this count from 0 or 1? if so do we need to -1 on the count?

File: ClimsoftVer4/ClimsoftVer4/ucrNavigation.vb [hunk 2]
```diff
@@ -1,4 +1,90 @@
...
+    Private Sub btnMoveLast_Click(sender As Object, e As EventArgs) Handles btnMoveLast.Click
+        'In order to move to move to the last record the record index is set to the maximum number of records minus one.
+        currRow = maxRows - 1
+        OnevtValueChanged(sender, e)
+
+    End Sub
+
+
 
 End Class
```
Comment (maxwellfundi) [inline ClimsoftVer4/ClimsoftVer4/ucrNavigation.vb]: Looks good, will have to test this on my dialog.

File: ClimsoftVer4/ClimsoftVer4/ucrNavigation.vb [hunk 3]
```diff
@@ -1,4 +1,112 @@
...
+        'Display the record number in the data navigation Textbox
+        If maxRows = 0 Then
+            txtRecNum.Text = "Record " & 0 & " of " & 0
+        Else
+            txtRecNum.Text = "Record " & currRow + 1 & " of " & maxRows
+        End If
+
+    End Sub
+
+    Protected Overridable Sub ucrNavigation_Load(sender As Object, e As EventArgs) Handles Me.Load
```
Comment (maxwellfundi) [inline ClimsoftVer4/ClimsoftVer4/ucrNavigation.vb:44]: In here we need to make the textbox for the navigation to be non editable- Actually maybe on an InitialiseControl sub

**Comment by maxwellfundi** [comment]
@Patowhiz  this control seems to work fine but then it has the same issue i have raised in one of the issues. 

I made a few edits though.

**Comment by Patowhiz** [comment]
Thanks for the test. The control is not finished yet. Have you seen how
instead of binding to the selectors we could bind to the navigation button?

**Comment by maxwellfundi** [comment]
@Patowhiz Do you want to push this here or you the all_dev_branch?

**Comment by Patowhiz** [comment]
Yep

**Comment by maxwellfundi** [comment]
@Patowhiz  I f i remember correctly you had merged this to the all dev branch that @dannyparsons got from my repo?

**Pull request merged**